File: src/com/android/k9/mail/MessagingException.java

package com.fsck.k9.mail;

public class MessagingException extends Exception {
    public static final long serialVersionUID = -1;
    
    public MessagingException(String message) {
        super(message);
    }

    public MessagingException(String message, Throwable throwable) {
        super(message, throwable);
    }
}

File: core/common/src/commonMain/kotlin/net/thunderbird/core/common/exception/MessagingException.kt
package net.thunderbird.core.common.exception

open class MessagingException(
    override val message: String?,
    private val isPermanentFailure: Boolean,
    override val cause: Throwable?,
) : Exception(message, cause) {

    constructor(cause: Throwable?) : this(message = null, cause = cause, isPermanentFailure = false)
    constructor(message: String?) : this(message = message, cause = null, isPermanentFailure = false)
    constructor(message: String?, isPermanentFailure: Boolean) : this(
        message = message,
        cause = null,
        isPermanentFailure = isPermanentFailure,
    )

    constructor(message: String?, cause: Throwable?) : this(
        message = message,
        cause = cause,
        isPermanentFailure = false,
    )

    companion object {
        private const val serialVersionUID = -1
    }
}
